### PR TITLE
Set `CMAKE_*_COMPILER_TARGET` to full triplet

### PR DESCRIPTION
**Pull request by longnguyen2004**
Leaving it as-is breaks clang faking as gcc, so set it to the full triplet.

### CODE REVIEW

**Review by nmoinvaz — APPROVED** [review]

**Comment by nmoinvaz** [comment]
Can you append " for mingw CMake toolchains" to the commit message?

**Comment by longnguyen2004** [comment]
Renamed!

File: cmake/toolchain-mingw-i686.cmake [hunk 1]
```diff
@@ -14,8 +14,8 @@ set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
 
 # Prefer posix gcc variant for gtest pthread support
 find_program(C_COMPILER_FULL_PATH NAMES
-    ${CMAKE_C_COMPILER_TARGET}-w64-mingw32-gcc-posix
-    ${CMAKE_C_COMPILER_TARGET}-w64-mingw32-gcc
+    ${CMAKE_C_COMPILER_TARGET}-gcc-posix
+    ${CMAKE_C_COMPILER_TARGET}-gcc
     gcc-mingw32-w64-${CMAKE_C_COMPILER_TARGET})
```
Comment (mtl1979) [inline cmake/toolchain-mingw-i686.cmake]: This like needs also be corrected...

File: cmake/toolchain-mingw-i686.cmake [hunk 2]
```diff
@@ -24,14 +24,14 @@ set(CMAKE_C_COMPILER ${C_COMPILER_FULL_PATH})
 
 find_program(CXX_COMPILER_FULL_PATH NAMES
     g++-mingw-w64-${CMAKE_CXX_COMPILER_TARGET}
```
Comment (mtl1979) [inline cmake/toolchain-mingw-i686.cmake]: This like needs also be corrected...

File: cmake/toolchain-mingw-i686.cmake (hunk 1 above)
Comment (longnguyen2004) [reply cmake/toolchain-mingw-i686.cmake]: I'm not aware of any distro that uses such a naming scheme, so I'll remove it.

File: cmake/toolchain-mingw-i686.cmake (hunk 2 above)
Comment (longnguyen2004) [reply cmake/toolchain-mingw-i686.cmake]: I'm not aware of any distro that uses such a naming scheme, so I'll remove it.

**Review by Dead2 — APPROVED** [review]
LGTM

**Pull request merged**